app/src/main/java/org/kiwix/kiwixmobile/webserver/WebServerHelper.java

```diff
@@ -0,0 +1,116 @@
+package org.kiwix.kiwixmobile.webserver;
+
+import android.util.Log;
+import androidx.annotation.NonNull;
+import java.net.InetAddress;
+import java.net.NetworkInterface;
+import java.net.SocketException;
+import java.util.ArrayList;
+import java.util.Enumeration;
+import org.kiwix.kiwixlib.JNIKiwixException;
+import org.kiwix.kiwixlib.JNIKiwixLibrary;
+import org.kiwix.kiwixlib.JNIKiwixServer;
+
+/**
+ * WebServerHelper class is used to set up the suitable environment i.e. getting the
+ * ip address and port no. before starting the WebServer
+ * Created by Adeel Zafar on 18/07/2019.
+ */
+
+public class WebServerHelper {
+  public static boolean isServerStarted;
+  private static int port;
+  private final JNIKiwixLibrary kiwixLibrary = new JNIKiwixLibrary();
+  private final JNIKiwixServer kiwixServer = new JNIKiwixServer(kiwixLibrary);
+  private static final String TAG = "WebServerHelper";
+
+  public WebServerHelper() {
+
+  }
+
+  public boolean startServerHelper(@NonNull ServerStateListener stateListener,
+      @NonNull ArrayList<String> selectedBooksPath) {
+
+    // 1. Get port from settings screen
+    // 2. Ask user to change port in settings if port is in use.
+    // OR
+    // Always use 8080 and when its not available then iterate this number.
+    String ip = getIpAddress();
+    ip = ip.replaceAll("\n", "");
+    if (ip.length() == 0) {
+      stateListener.serverFailed();
+    } else if (!isServerStarted && startAndroidWebServer(selectedBooksPath)) {
+      stateListener.serverStarted("http://" + ip + ":" + port);
+    }
+    return isServerStarted;
+  }
+
+  public boolean stopAndroidWebServer(@NonNull ServerStateListener stateListener) {
+    if (isServerStarted) {
+      kiwixServer.stop();
+      isServerStarted = false;
+      stateListener.serverStopped();
+      return true;
+    }
+    return false;
+  }
+
+  private boolean startAndroidWebServer(ArrayList<String> selectedBooksPath) {
+    if (!isServerStarted) {
+      port = 8080;
+      //Call to start server
+      for (String path : selectedBooksPath) {
+        try {
+          boolean isBookAdded = kiwixLibrary.addBook(path);
+          Log.v(TAG, "isBookAdded: " + isBookAdded + path);
+        } catch (JNIKiwixException e) {
+          Log.v(TAG, "Couldn't add book " + path);
+        }
+      }
+      kiwixServer.setPort(port);
+      isServerStarted = kiwixServer.start();
+      Log.v(TAG, "Server status" + isServerStarted);
+    }
+    return isServerStarted;
+  }
+
+  // get Ip address of the device's wireless access point i.e. wifi hotspot OR wifi network
+  private static String getIpAddress() {
```

Comment:
shouldn't need to be static, it is private after all

Reply:
Fixed in 15b66755b1f1b1ab1f15db5e494e6359a50495ba